### ISSUE TITLE
Turn detector MultilingualModel: add Romanian (ro) language support

### ISSUE THREAD

**Issue by arsenieciprian** [livekit/agents#5159]
### Feature Type

Would make my life easier

### Feature Description

Hi LiveKit team,                                                                                                                                                                                                                               
                  
  We're running a production voice agent for Romanian telephony                                                                                                                                                                                  
  (inbound/outbound calls) using LiveKit Agents 1.4.6.                                                                                                                                                                                           
                                                                                                                                                                                                                                                 
  Currently MultilingualModel (v0.4.1-intl) supports 14 languages                                                                                                                                                                                
  but Romanian (ro) is not among them. We verified languages.json                                                                                                                                                                                
  directly from the model repo.                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                 
  Romanian is a Romance language (similar to Italian/Portuguese
  which are already supported), so we'd expect training data                                                                                                                                                                                     
  to be available.                                          
                                                                                                                                                                                                                                                 
  Use case: AI receptionist for a notary office in Romania,
  handling appointment scheduling via phone calls.                                                                                                                                                                                               
                                                  
  Would it be possible to add ro to the next multilingual release?                                                                                                                                                                               
  Happy to help with testing/validation data if needed.                                                                                                                                                                                          
                                                                                                                                                                                                                                                 
  Thanks                                                                                                                                                                                                                                         

### Workarounds / Alternatives

_No response_

### Additional Context

_No response_

**Comment by tinalenguyen** [comment]
hi, thank you for the request! we'll keep this in mind for future iterations

**Comment by arsenieciprian** [comment]
thank you

**Comment by arsenieciprian** [comment]
Hi team, just a gentle ping on this one — Romanian language support for MultilingualModel would be very useful for our production voice agent. Any chance it makes it into an upcoming release? Happy to help with testing if needed. Thanks!

**Comment by arsenieciprian** [comment]
Following up again — this has become a real quality problem for us, so I want to add concrete detail rather than just ping.

We run a production telephony voice agent (LiveKit Agents JS, cascaded STT→LLM→TTS) that takes taxi orders in Romanian. Since `MultilingualModel` has no `ro`, we are stuck on VAD + fixed endpointing thresholds, and that tradeoff has no good setting:

- **Low thresholds** → the agent cuts customers off mid-address. Addresses are exactly where people pause to think ("strada… ăăă… Aviatorilor, numărul…"). A cut there means a *wrong address on a real taxi order*, which is the worst failure we can have.
- **High thresholds** → we currently run `minDelay=1500ms` / `maxDelay=2200ms`, raised deliberately after exactly those cut-offs. The agent is now safe but audibly sluggish on every single turn, including the ones where the caller obviously finished.

A blind timer cannot tell "strada…" (unfinished) from "…Aviatorilor 5" (finished) — which is precisely what a semantic turn detector solves. There is no workaround available to us: no threshold value fixes both cases, and the language is not something we can change.

I understand `ro` is a smaller market and the model covers the big ones first. But the gap is not cosmetic — for a telephony agent it is the difference between a usable product and one that either interrupts customers or feels slow on every turn. If it helps, I am happy to contribute Romanian evaluation data or test builds against real production traffic — offer still stands from April.

Any indication of whether additional languages are on the roadmap at all would help us plan. Even a "not planned" is more useful to us than uncertainty, since it would tell us to stop waiting and look at alternatives.

**Comment by arsenieciprian** [comment]
Following up with measurements instead of just asking — we evaluated `v0.4.1-intl` on Romanian using our own production data, to check whether the model might already generalize (Romanian is a Romance language, and it/es/fr/pt are supported). **It does not.**

### Setup

- Model: `livekit/turn-detector`, revision `v0.4.1-intl`, `onnx/model_q8.onnx` — i.e. exactly what the plugin pins today.
- Inference path replicated 1:1 from `agents-plugin-livekit`: `normalizeText` → `apply_chat_template(add_generation_prompt=false)` → slice at the last `<|im_end|>` → take `prob[-1]`. Sanity-checked against the plugin behaviour on English.
- Data: 1,231 real Romanian telephony calls (taxi dispatch), 300 evaluation pairs.
- Each pair = one real user turn, in its real conversation context (≤6 turns), in two forms:
  - **positive**: the complete turn (STT finalized, agent responded)
  - **negative**: the same turn truncated immediately after a preposition / article / address anchor ("de pe", "la", "strada", "blocul"), so it is unambiguously unfinished regardless of context.

### Results

| | AUC | median P(complete) | median P(unfinished) | TPR / TNR @ `it` threshold |
|---|---|---|---|---|
| **Romanian** | **0.594** | 0.0316 | 0.0143 | 82% / 31% |
| **English** (control, same code, same model) | **0.931** | 0.2692 | 0.0004 | 92% / 83% |

On English the two classes separate by ~3 orders of magnitude. On Romanian the distributions overlap almost entirely — **AUC 0.594 is close to chance (0.5)**.

Borrowed thresholds do not rescue it: at `it`/`es`/`fr`/`pt`/`en` thresholds we get TPR 70–82% at TNR 31–47%. Calibrating a `ro` threshold the way `languages.json` appears to be calibrated (TPR ≈ 99.3%) drives the threshold to ~0 and **TNR to 1%** — it would fire on essentially every mid-sentence pause.

Failures are not subtle, they are inverted. Representative shape (paraphrased, not real customer data):

```
complete:   "from Berceni, block 23A, staircase A"    p = 0.0026
unfinished: "from Berceni, block 23A, staircase"      p = 0.0958   <- 37x more "finished"
```

### Caveats, stated honestly

- Negatives are synthetic truncations of real utterances, not naturally occurring cut-offs.
- The English control is 12 hand-written taxi-style pairs, not production data — it validates the harness, it is not a benchmark.
- Single domain (taxi dispatch: addresses, names, numbers), which is admittedly the hardest case for EOU.

### Why this matters

The practical takeaway is that **`ro` cannot be added by picking a threshold** — the underlying model does not separate the classes at all, so there is nothing to calibrate. It needs actual training data.

Two things that would genuinely unblock us, in order of preference:

1. **Add `ro` to the multilingual model.** We are happy to contribute Romanian evaluation data and to test candidate builds against live production traffic.
2. **Failing that, publish the training/fine-tuning recipe** (#3183 was closed without discussion). The weights are open and the licence permits derivative works used with LiveKit Agents, so the only thing missing is the recipe. We have ~7.5k in-domain Romanian turns and would do the work ourselves.

Happy to share the evaluation script if useful.

**Comment by bhargavikalicheti** [comment]
Hi @arsenieciprian, I'd be interested in working on Romanian turn-detection support. I noticed that the original request targets MultilingualModel, while the turn-detector plugin is now deprecated in favor of livekit.agents.inference.TurnDetector.

@livekit maintainers — is Romanian support still needed for the new TurnDetector architecture? If so, is language expansion something that can be contributed through this repository, and could you point me toward the preferred implementation/evaluation path?

I'd be happy to work on the implementation, tests, and Romanian turn-detection validation if this is still something you'd like to support.

**Comment by arsenieciprian** [comment]
Hi @bhargavikalicheti, thank you — Romanian support is still very much needed on our side (production telephony voice agents, taxi dispatch).

We took the architecture pointer seriously and evaluated the **new** `TurnDetector` on Romanian before answering. The local `v1-mini` path is audio-based and language-agnostic at inference — `EOT.predict()` takes only the last 1.2 s of 16 kHz PCM, and language merely selects the decision threshold from `inference/eot/languages.py` (unlisted languages fall back to the `en` default and still run). So the question became: *can Romanian be added by just calibrating a `ro` threshold?* We measured it. **It cannot.**

### Setup

- Exact shipped model: `livekit-local-inference` 0.2.6, `EOT.predict`, 1.2 s windows ending 200 ms into silence (matching `MIN_SILENCE_DURATION_MS`).
- Data: 1,409 real Romanian telephony calls (stereo recordings, customer channel isolated), VAD segmentation with our production settings.
- Labels from dialogue reality: **negative** = customer resumes within ≤1.2 s with no agent reply (hesitation pause, unfinished); **positive** = the agent takes the turn. Ambiguous points dropped.

### Results (1,242 calls, 7,695 points)

| | value |
|---|---|
| AUC | **0.565** (0.5 = chance) |
| median P, completed turns | 0.371 |
| median P, hesitation pauses | 0.319 |
| TPR / TNR @ `en` default 0.36 | 52% / 57% |
| best operating point | Youden J = 0.089 |

Sanity probes on the same audio confirm the harness: the model clearly responds to the input — P rises with trailing silence (0.37 → 0.42 → 0.54 at +0.2/+0.8/+3.0 s) — but mid-speech windows get a median of 0.30, i.e. on Romanian narrowband audio it behaves closer to a silence-duration estimator than a turn-completion detector.

**Caveats, stated honestly:** labels are naturalistic and therefore noisy (some positives were force-committed by our fixed endpointing; some negatives may be completed turns followed by afterthoughts); audio is 8 kHz telephony upsampled to 16 kHz, decoded from mp3; we had no comparable English telephony corpus for a control this time. Noisy labels depress AUC toward 0.5, but they cannot hide a real signal this completely — and the mid-speech behavior is label-independent.

### Takeaway

Same shape as our earlier result on the deprecated text model (AUC 0.594): **Romanian support for the new architecture also cannot be added by picking a threshold — the model needs to see Romanian in training.** Which makes @bhargavikalicheti's offer land on the training side, not calibration.

@livekit maintainers — the two questions that would unblock this:

1. Is language expansion for `turn-detector-v1` / `v1-mini` something external contributors can participate in (training data, evaluation, testing candidate builds)?
2. Failing that, would you publish the training/fine-tuning recipe? (#3183 was closed without discussion.)

We can contribute Romanian evaluation pairs from ~7.5k in-domain turns and validate candidate builds against live production traffic; the evaluation harness used above is available on request.

**Comment by arsenieciprian** [comment]
Update with a proper control, now that eot-bench is public.

We validated our Romanian evaluation against LiveKit's own benchmark: our scoring reproduces the committed `v1-mini` predictions for `en` exactly (1,105 spans, correlation 1.0000). With the identical method, `v1-mini` scores AUC 0.79–0.89 on all 14 benchmark languages (0.887 for English even after 8 kHz band-limiting + G.711 µ-law, so telephony is not the cause) — and **0.565 on 1,242 real Romanian calls**, i.e. near chance. Language-agnostic at inference does not mean it generalizes to an untrained language.

Full numbers and the ask (a `ro` split in `eot-bench-data` + Romanian in the next training round) are in livekit/eot-bench#14. We can re-run any candidate model on our Romanian set and report aggregated results.
